python_modules/dagster/dagster/core/execution.py

```diff
@@ -576,11 +630,26 @@ def _create_resources(pipeline_def, context_def, environment, execution_context,
     # can potentially have many resources so we need to use this abstraction.
     with ExitStack() as stack:
         for resource_name in context_def.resources.keys():
-            resource_obj_or_gen = get_resource_or_gen(
-                pipeline_def, context_def, resource_name, environment, run_id
+
+            resource_def = context_def.resources[resource_name]
+            # Need to do default values
+            resource_config = environment.context.resources.get(resource_name, {}).get('config')
+            init_context = InitResourceContext(
+                pipeline_def=pipeline_def,
+                resource_def=resource_def,
+                context_config=environment.context.config,
+                resource_config=resource_config,
+                run_id=run_id,
             )
 
-            resource_obj = stack.enter_context(as_ensured_single_gen(resource_obj_or_gen))
+            # capture loop vars as default params to lambda
```

Comment:
hmmm  i think you are going to have to explain this one

Reply:
https://stackoverflow.com/questions/2295290/what-do-lambda-function-closures-capture

Reply:
Ah I see. I generally create a function that returns a lambda as I think that is a lot more explicit. 

```
def create_resource_lambda(resource_def, init_context):
    return lambda: resource_def.resource_fn(init_context)
```

However we can do this default variable trick too. Just seems a little clever to me